legend-engine-xts-data-space/legend-engine-xt-data-space-compiler/src/main/java/org/finos/legend/engine/language/pure/compiler/toPureGraph/DataSpaceCompilerExtension.java

```diff
@@ -159,11 +138,43 @@ public Iterable<? extends Processor<?>> getExtraProcessors()
                     }
 
                     // executables
+                    HashSet<String> executableTitles = new HashSet<>();
                     metamodel._executables(dataSpace.executables != null ? ListIterate.collect(dataSpace.executables, executable ->
-                            new Root_meta_pure_metamodel_dataSpace_DataSpaceExecutable_Impl("", null, context.pureModel.getClass("meta::pure::metamodel::dataSpace::DataSpaceExecutable"))
-                                    ._title(executable.title)
-                                    ._description(executable.description)
-                                    ._executable(context.pureModel.getPackageableElement(executable.executable.path, executable.executable.sourceInformation))) : null);
+                    {
+                        if (executableTitles.add(executable.title))
+                        {
+                            if (executable instanceof DataSpacePackageableElementExecutable)
+                            {
+                                return new Root_meta_pure_metamodel_dataSpace_DataSpacePackageableElementExecutable_Impl("", null, context.pureModel.getClass("meta::pure::metamodel::dataSpace::DataSpacePackageableElementExecutable"))
+                                        ._title(executable.title)
+                                        ._description(executable.description)
+                                        ._executable(context.pureModel.getPackageableElement(((DataSpacePackageableElementExecutable) executable).executable.path, ((DataSpacePackageableElementExecutable) executable).executable.sourceInformation));
+                            }
+                            else if (executable instanceof DataSpaceTemplateExecutable)
+                            {
+
+
+                                    if (((DataSpaceTemplateExecutable) executable).executionContextKey != null && !dataSpace.executionContexts.stream().map(c -> c.name).collect(Collectors.toList()).contains(((DataSpaceTemplateExecutable) executable).executionContextKey))
+                                    {
+                                        throw new EngineException("Data space template executable's executionContextKey is not valid", dataSpace.sourceInformation, EngineErrorType.COMPILATION);
```

Comment:
we providing the correct sourceinformation for these errors? 
they should be on the executables source information right?